index.js

```diff
@@ -3,26 +3,38 @@ const path = require('path');
 
 const pluginName = 'Chunks2JsonWebpackPlugin';
 
+const defaultOptions = {
+    // ignore files emitted by HMR by default
+    excludeFile: /\.hot-update\.js$/,
+    // group chunks by extension
+    chunkGroupName: filename => /\.([a-z0-9]+(\.map)?)(\?.*)?$/.exec(filename)[1],
+    outputDir: process.cwd(),
+    filename: 'build-manifest.json',
+    // generate contents to save to manifest file
+    objectToString: result => JSON.stringify(result),
+    publicPath: ''
+};
+
 class Chunks2JsonWebpackPlugin {
     constructor(options) {
-        this.options = options;
+        this.options = Object.assign({}, defaultOptions, options);
         this.result = {};
     }
     apply(compiler) {
         compiler.hooks.emit.tap(pluginName, compilation => {
-            compilation.chunks.forEach((chunk) => {
-                if (this.result[chunk.name] === undefined) {
+            this.result = {};
+            compilation.chunks.forEach(chunk => {
+                if (!this.result[chunk.name]) {
                     this.result[chunk.name] = {};
                 }
-                chunk.files.forEach((filename) => {
-                    if (filename.endsWith('css')) {
-                        this.result[chunk.name].css = `/${filename}`;
-                    } else if (filename.endsWith('js')) {
-                        this.result[chunk.name].js = `/${filename}`;
-                    } else if (filename.endsWith('js.map')) {
-                        this.result[chunk.name].jsMap = `/${filename}`;
-                    } else if (filename.endsWith('css.map')) {
-                        this.result[chunk.name].cssMap = `/${filename}`;
+                chunk.files.forEach(filename => {
+                    const exclude = typeof this.options.excludeFile === 'function' ?
+                        this.options.excludeFile(filename, chunk) :
+                        this.options.excludeFile.test(filename);
+                    if (!exclude) {
```

Comment:
Here we have a situation with nested `if`, which is hard to read. I recommend in general to avoid nesting conditions and if possible avoid using `else`. Please format this to exit the function if exit condition is meet and continue otherwise. 

I suggest that: 

```javascript
const exclude = typeof this.options.excludeFile === 'function' ?
  this.options.excludeFile(filename, chunk) :
  this.options.excludeFile.test(filename);
  if (exclude === true) {
     // from that point on, we don't care what happens, we just know we want to ignore it, so we break from the function
     return;
  }
  // otherwise we continue with the flow
  const ext = this.options.chunkGroupName(filename, chunk);
  ......
```

Also ternary could be abstracted to a function `shouldIgnoreFile()`, which would again increase readability (this is optional though, as one-liner can stay as is for now, until it becomes more complicated)